worker/worker.go

```diff
@@ -649,14 +624,26 @@ func (w *worker) rhpFormHandler(jc jape.Context) {
 
 		contract, txnSet, err = RPCFormContract(ctx, t, renterKey, renterTxnSet)
 		if err != nil {
-			w.bus.WalletDiscard(ctx, renterTxnSet[len(renterTxnSet)-1])
+			ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
This could be a helper method on the worker called `discardTxn` maybe. Would be nice because we are forgetting to handle the context probably in contract formations as well.